src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -517,20 +517,33 @@ public void AddMetadataReference(MetadataReference reference, AssemblyIdentity i
             }
         }
 
-        private static readonly string s_GAC = $"{Path.DirectorySeparatorChar}GAC_MSIL{Path.DirectorySeparatorChar}";
-        private static readonly string s_ALTGAC = $"{Path.AltDirectorySeparatorChar}GAC_MSIL{Path.AltDirectorySeparatorChar}";
-
         private bool IsInGAC(string filePath)
         {
-            return filePath.Contains(s_GAC) || filePath.Contains(s_ALTGAC);
+            return GlobalAssemblyCacheLocation.RootLocations.Any(gloc => FilePathUtilities.IsNestedPath(gloc, filePath));
```

Comment:
Looking at the implementation, this will fail on Mono currently. We have code to find the Mono GAC here: http://source.roslyn.io/#Microsoft.CodeAnalysis.Scripting/Resolvers/MonoGlobalAssemblyCache.cs